src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine_MapCreation.cs

```diff
@@ -165,16 +165,23 @@ internal partial class FindReferencesSearchEngine
                 {
                     finderTasks.Add(Task.Run(async () =>
                     {
-                        var symbols = await f.DetermineCascadedSymbolsAsync(searchSymbol, _solution, projects, _cancellationToken).ConfigureAwait(false) ?? SpecializedCollections.EmptyEnumerable<ISymbol>();
+                        var symbolTasks = new List<Task>();
 
-                        _cancellationToken.ThrowIfCancellationRequested();
+                        var symbols = await f.DetermineCascadedSymbolsAsync(searchSymbol, _solution, projects, _cancellationToken).ConfigureAwait(false);
+                        AddSymbolTasks(result, symbols, symbolTasks);
 
-                        List<Task> symbolTasks = new List<Task>();
-                        foreach (var child in symbols)
+                        // Defer to the language to see if it wants to cascade here in some special way.
+                        var symbolProject = _solution.GetProject(searchSymbol.ContainingAssembly);
```

Comment:
So the normal cascading from metadata-to-source etc. is already happening through the overall FAR logic.  i.e. 'searchSymbol' is not the first symbol.  It's whatever symbol we've cascaded to to that we're now using to fan out from.  This part here simply says "once we have a symbol we've decided ot check, and it's a source symbol, see if the language owning it's defining project has any extra cascades it may want to perform.